platform/android/src/style/sources/geojson_source.cpp

```diff
@@ -125,6 +110,55 @@ namespace android {
         return jni::Object<Source>(GeoJSONSource::javaClass.New(env, constructor, reinterpret_cast<jni::jlong>(this)).Get());
     }
 
+    template <class JNIType>
+    void GeoJSONSource::setCollectionAsync(jni::JNIEnv& env, jni::Object<JNIType> jObject) {
+
+        std::shared_ptr<jni::jobject> object = std::shared_ptr<jni::jobject>(jObject.NewGlobalRef(env).release()->Get(), GenericGlobalRefDeleter());
```

Comment:
🤔 Did you also consider just using `std::make_shared<jni::UniqueObject>`? That way you wouldn't have to worry about the release/get/custom-deleter logic because `jni::UniqueObject` would take care of it all. I know it's a little weird to have a shared_ptr to a unique_ptr, but I can't think of a real problem with doing it that way...

Reply:
Capturing from a convo that we can't use `std::make_shared` here as we need to pass a custom destructor.

Reply:
I think specifically what's happening is that `shared_ptr` has a move contructor that takes a `unique_ptr` as an argument (see #13 at https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr). Anyway, your current approach seems fine.